### PR TITLE
Setup of Mongodb

### PR DESCRIPTION
**Pull request by Phil9219**

### CODE REVIEW

File: lib/postRun.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const { collection } = require("./database");
+
+async function postRun(runName, distance, startDate, pace) {
+  const newrun = await collection("rundata").insertOne({
+    runName,
+    distance,
+    startDate,
+    pace,
+  });
```
Comment (Naasgul) [inline lib/postRun.js]: I'm not too sure, but isn't "insertOne" used for inserting just one argument? Maybe you need to change it to just ".insert"

Comment (Phil9219) [reply lib/postRun.js]: Issue is solved. 👍

File: lib/searchForKm.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const { collection } = require("./database");
+
+async function searchForKm(distance) {
+  console.log(distance);
+  const runs = await collection("rundata")
+    // .find({ distance: { $lte: distance } })
+    .find({ distance: distance })
```
Comment (mhstrkmp) [inline lib/searchForKm.js]:
```suggestion
    .find({ distance: { $lte: `${distance}` })
```

I think that should fix the filter problem ...

Comment (Phil9219) [reply lib/searchForKm.js]: Unfortunately still not.

File: client/src/api/runs.js [hunk 3]
```diff
@@ -10,17 +10,19 @@ export async function getRunsById(id) {
...
 export async function postRun(run) {
-  const response = await fetch("/api/runs", {
+  console.log(run);
+  const response = await fetch("/api/runs/", {
     method: "POST",
     body: JSON.stringify(run),
     headers: { "Content-Type": "application/json" },
   });
   const newRun = await response.json();
+  console.log(newRun);
```
Comment (ncbecker) [inline client/src/api/runs.js]: Remove before merge

File: client/src/components/Page/ChooseYourRunPage.js [hunk 4]
```diff
@@ -33,6 +34,7 @@ export default function ChooseYourRunPage() {
     event.preventDefault();
     if (searchValue) {
       setRuns(await searchForKm(searchValue));
+      console.log(searchValue);
```
Comment (ncbecker) [inline client/src/components/Page/ChooseYourRunPage.js]: Remove before merge

File: client/src/components/Page/CreatedRunPage.js [hunk 5]
```diff
@@ -7,15 +7,16 @@ import { useParams } from "react-router-dom";
 import { CreatedRunContainer } from "../../components/PageContainer";
 
 export default function CreatedRunPage() {
-  const { id } = useParams();
-  const [run, setRun] = useState(id);
+  const { _id } = useParams();
+  console.log(_id);
```
Comment (ncbecker) [inline client/src/components/Page/CreatedRunPage.js]: Remove before merge

File: client/src/components/Page/ChooseYourRunPage.js [hunk 6]
```diff
@@ -33,6 +34,7 @@ export default function ChooseYourRunPage() {
     event.preventDefault();
```
Comment (ncbecker) [inline client/src/components/Page/ChooseYourRunPage.js]: If this is not a submit function for your form, I think you can remove it

File: lib/database.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+let db;
+
+async function connect(url, dbName) {
+  client = await MongoClient.connect(url, { useUnifiedTopology: true });
+  db = client.db(dbName);
+}
+
+function close() {
+  return client.close();
+}
```
Comment (ncbecker) [inline lib/database.js]: Do you use this function?

File: lib/getRuns.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+const { collection } = require("./database");
+
+async function getRuns() {
+  const runs = await collection("rundata")
+    .find({})
+    .sort({ startDate: 1 })
+    .toArray();
+
+  if (!runs) {
+    console.log("No Runs Available");
```
Comment (ncbecker) [inline lib/getRuns.js:10]: Maybe send this message as server response not console.log?

File: lib/getRunsById.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
+const { collection } = require("./database");
+const { ObjectId } = require("mongodb");
+async function getRunsById(_id) {
+  const objectId = new ObjectId.createFromHexString(_id);
+  const runs = await collection("rundata").findOne({ _id: objectId });
+
+  console.log(runs);
```
Comment (ncbecker) [inline lib/getRunsById.js]: Remove before merge

File: lib/searchForKm.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
+const { collection } = require("./database");
+
+async function searchForKm(km) {
+  const runs = await collection("rundata")
+    .find({ distance: { $lte: parseFloat(km) } })
+    .sort({ distance: -1 })
+    .toArray();
+  console.log(runs);
```
Comment (ncbecker) [inline lib/searchForKm.js]: Remove before merge

File: server.js [hunk 11]
```diff
@@ -14,14 +22,63 @@ app.use(
...
+  } catch (error) {
+    console.error(error);
+    res.status(500).send("Unexpected server error");
+  }
+});
+app.get("/api/runs/:_id/", async (req, res) => {
+  const { _id } = req.params;
+  try {
+    const allRuns = await getRunsById(_id);
+    res.json(allRuns);
```
Comment (ncbecker) [inline server.js]: Maybe also send status code 200 in this response?

**Review by ncbecker — APPROVED** [review]
lgtm 👍 some logs that should be removed before merging

File: client/src/components/Page/ChooseYourRunPage.js (hunk 6 above)
Comment (Phil9219) [reply client/src/components/Page/ChooseYourRunPage.js]: FYI 
Kept it in, so the page will not reload each time I submit the search.

**Pull request merged**